### PR TITLE
Implement scoring for the SR2023 game

### LINKED ISSUE #1040: Write scoring scripts for the SR2023 game
The scoring script provides automatic computation of the scores based on the data input from the score sheets.
See https://github.com/PeterJCLaw/srcomp/wiki/Competition-Preparation#implement-scorepy for details.

### Original

[comp/game/score-script](https://github.com/srobo/recurring-tasks/blob/master/comp/game/score-script.yaml)

### Dependencies

 * [ ] #1039 Create SR2023 game score format

### PR DESCRIPTION
**Pull request by PeterJCLaw**
Fixes https://github.com/srobo/tasks/issues/1040

### CODE REVIEW

**Comment by PeterJCLaw** [comment]
CI failure is due to lack of matches.

File: scoring/score.py [hunk 1]
```diff
@@ -8,7 +28,63 @@ def __init__(self, teams_data, arena_data):
...
+                raise InvalidScoresheetException(
+                    f"Too few {token_type} tokens seen. "
+                    f"Must be at least {min_count} got {count}",
+                )
+
+        missing_but_moving_teams = [
+            tla
+            for tla, info in self._teams_data.items()
+            if info.get('left_scoring_zone', False)
+            if not info.get('present', True)
```
Comment (sedders123) [inline scoring/score.py:80]: Should we be assuming that teams are present if the data is missing? (Though this data should never be missing so don't feel too strongly about this)

File: scoring/tests/test_scoring.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+            'ABC': {'zone': 0, 'present': True},
+            'DEF': {'zone': 1, 'present': True},
+        }
+        self.assertScores(
+            {'ABC': 81, 'DEF': 81},
+            {},
+            self.zone_tokens,
+        )
+
+    def test_left_scoring_zone_but_absent(self):
```
Comment (sedders123) [inline scoring/tests/test_scoring.py:232]: Does this need its own section? Not really a tolerable input deviance

**Review by sedders123 — APPROVED** [review]
Looks good to me! Just a couple of small nit picky comments.

I'm also assuming that disqualification is handled by a different part of SRComp and when calculating scores we don't care about disqualification.

File: scoring/score.py (hunk 1 above)
Comment (PeterJCLaw) [reply scoring/score.py:80]: > Should we be assuming that teams are present if the data is missing?

The logic here is copied from previous years. Note however that the behaviour of the `present` and `disqualified` when missing are defined by the Proton spec for score files: https://github.com/PeterJCLaw/proton#inputs

File: scoring/tests/test_scoring.py (hunk 2 above)
Comment (PeterJCLaw) [reply scoring/tests/test_scoring.py:232]: Ah, good spot.

Comment (PeterJCLaw) [reply scoring/tests/test_scoring.py:232]: 8f80987

**Pull request merged**